runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingPlugin.java

```diff
@@ -243,49 +242,28 @@ public void onInteractingChanged(InteractingChanged event)
 		currentSpot = spot;
 	}
 
-	private boolean canPlayerFish(final ItemContainer itemContainer)
+	private boolean canPlayerFish(final ItemContainer... itemContainer)
 	{
-		if (itemContainer == null)
+		usableGear.clear();
+		for (ItemContainer container : itemContainer)
 		{
-			return false;
-		}
+			if (container == null)
+			{
+				continue;
+			}
 
-		for (Item item : itemContainer.getItems())
-		{
-			switch (item.getId())
+			for (Item item : container.getItems())
 			{
-				case ItemID.DRAGON_HARPOON:
-				case ItemID.DRAGON_HARPOON_OR:
-				case ItemID.INFERNAL_HARPOON:
-				case ItemID.INFERNAL_HARPOON_UNCHARGED:
-				case ItemID.INFERNAL_HARPOON_UNCHARGED_25367:
-				case ItemID.HARPOON:
-				case ItemID.BARBTAIL_HARPOON:
-				case ItemID.BIG_FISHING_NET:
-				case ItemID.SMALL_FISHING_NET:
-				case ItemID.SMALL_FISHING_NET_6209:
-				case ItemID.FISHING_ROD:
-				case ItemID.FLY_FISHING_ROD:
-				case ItemID.PEARL_BARBARIAN_ROD:
-				case ItemID.PEARL_FISHING_ROD:
-				case ItemID.PEARL_FLY_FISHING_ROD:
-				case ItemID.BARBARIAN_ROD:
-				case ItemID.OILY_FISHING_ROD:
-				case ItemID.LOBSTER_POT:
-				case ItemID.KARAMBWAN_VESSEL:
-				case ItemID.KARAMBWAN_VESSEL_3159:
-				case ItemID.CORMORANTS_GLOVE:
-				case ItemID.CORMORANTS_GLOVE_22817:
-				case ItemID.INFERNAL_HARPOON_OR:
-				case ItemID.TRAILBLAZER_HARPOON:
-				case ItemID.CRYSTAL_HARPOON:
-				case ItemID.CRYSTAL_HARPOON_23864:
-				case ItemID.CRYSTAL_HARPOON_INACTIVE:
-					return true;
+				FishingTools toolType = FishingTools.getToolType(item.getId());
+				if (toolType != null)
+				{
+					usableGear.add(toolType);
```

Comment:
This method should not mutate state. Track gear changes in `ItenContainerChanged` events, not through this method.
```suggestion
					return true;
```